oadp-operator/controllers/velero.go

```diff
@@ -1,14 +1,469 @@
 package controllers
 
 import (
+	"fmt"
 	"github.com/go-logr/logr"
 	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"k8s.io/utils/pointer"
+)
+
+const (
+	Velero          = "velero"
+	VeleroNamespace = "oadp-operator"
+	//TODO: Check for default secret names
+	VELERO_AWS_SECRET_NAME   = "cloud-credentials"
+	VELERO_AZURE_SECRET_NAME = "cloud-credentials-azure"
+	VELERO_GCP_SECRET_NAME   = "cloud-credentials-gcp"
+)
+
+// Environment Vars
+const (
+	LD_LIBRARY_PATH               = "LD_LIBRARY_PATH"
+	VELERO_NAMESPACE              = "VELERO_NAMESPACE"
+	VELERO_SCRATCH_DIR            = "VELERO_SCRATCH_DIR"
+	AWS_SHARED_CREDENTIALS_FILE   = "AWS_SHARED_CREDENTIALS_FILE"
+	AZURE_SHARED_CREDENTIALS_FILE = "AZURE_SHARED_CREDENTIALS_FILE"
+	GCP_SHARED_CREDENTIALS_FILE   = "GCP_SHARED_CREDENTIALS_FILE"
+	//TODO: Add Proxy env vars
+	HTTP_PROXY  = "HTTP_PROXY"
+	HTTPS_PROXY = "HTTPS_PROXY"
+	NO_PROXY    = "NO_PROXY"
+)
+
+//TODO: Add Image customization options
+// Images
+const (
+	VELERO_IMAGE           = "quay.io/konveyor/velero:latest"
+	OPENSHIFT_PLUGIN_IMAGE = "quay.io/konveyor/openshift-velero-plugins:latest"
+	AWS_PLUGIN_IMAGE       = "quay.io/konveyor/velero-plugin-for-aws:latest"
+	AZURE_PLUGIN_IMAGE     = "quay.io/konveyor/velero-plugin-for-microsoft-azure:latest"
+	GCP_PLUGIN_IMAGE       = "quay.io/konveyor/velero-plugin-for-gcp:latest"
+	CSI_PLUGIN_IMAGE       = "quay.io/konveyor/velero-plugin-for-csi:latest"
+)
+
+// Plugin names
+const (
+	VELERO_PLUGIN_FOR_AWS       = "velero-plugin-for-aws"
+	VELERO_PLUGIN_FOR_AZURE     = "velero-plugin-for-microsoft-azure"
+	VELERO_PLUGIN_FOR_GCP       = "velero-plugin-for-gcp"
+	VELERO_PLUGIN_FOR_CSI       = "velero-plugin-for-csi"
+	VELERO_PLUGIN_FOR_OPENSHIFT = "openshift-velero-plugin"
 )
 
 func (r *VeleroReconciler) ReconcileVeleroDeployment(log logr.Logger) (bool, error) {
 	velero := oadpv1alpha1.Velero{}
 	if err := r.Get(r.Context, r.NamespacedName, &velero); err != nil {
 		return false, err
 	}
+
+	// Build Velero Deployment
+	veleroDeployment := r.buildVeleroDeployment(&velero)
+
+	// Create or Update Velero Deployment
+	op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, &veleroDeployment, func() error {
+
+		// Set controller reference to Velero controller
+		err := controllerutil.SetControllerReference(&velero, &veleroDeployment, r.Scheme)
```

Comment:
I don't think that this is changing anything in the cluster just on the object reference